File: Medium/Loop/3932. Count K-th Roots in a Range.py
# 3932. Count K-th Roots in a Range ->python
class Solution:
    def countKthRoots(self, l: int, r: int, k: int) -> int:

        if(k == 1): return r-l+1

        i = 0
        while((i ** k) < l): i+=1
        if((i ** k) > r): return 0

        j = i+1

        while((j ** k) <= r): j+=1
        return j-i
        

File: Medium/Loop/3932. Count K-th Roots in a Range.cpp
//3932. Count K-th Roots in a Range ->c++
//app-> 1 -> loop
class Solution {
public:
    int countKthRoots(int l, int r, int k) {
        if(k == 1) return r-l+1;

        int i = 0;
        while(pow(i,k) < l) i++;
        if(pow(i,k) > r) return 0;

        int j = i+1;

        while(pow(j,k) <= r) j++;
        return j-i;
  
    }
};
